sphinx/texinputs/sphinxlatexcontainers.sty

```diff
@@ -0,0 +1,17 @@
+%% CONTAINER DIRECTIVES
+%
+% change this info string if making any custom modification
+\ProvidesFile{sphinxlatexcontainers.sty}[2021/05/03 containers]
+
+% The purpose of this file is to provide a dummy environment sphinxclass
+% which will be inserted for each class in each container directive. The class
+% name will be passed as the argument to the environment. User-defined
+% formatting of directives can be achieved by renewing the definition of this
+% environment, and branching on the value of the argument.
+
+\ifx\sphinxclass\undefined % poor man's "provideenvironment"
+    \newenvironment{sphinxclass}[1]{
+        \def\sphinxClassFunctionName{sphinxCLASS#1}%
+        \csname \sphinxClassFunctionName \endcsname}%
+    {\csname end\sphinxClassFunctionName \endcsname}%
```

Comment:
Actually what is important is that both `\begin` and `\end` are used in exactly the same way. In the quoted code from sphinxlatexadmonitions.sty, if `\begin{sphinx\spx@noticetype}` was used then it would have to be `\end{shinx\spx@noticetype}` and it would work too.

In our case we either do not use any `\expandafter` or if we use them, we use them both for `\begin` and `\end`.

Reply:
OK. I believe I have implemented your first suggestion.